lib/Hakyll/Core/Compiler/Internal.hs

```diff
@@ -196,7 +196,12 @@ instance Applicative Compiler where
 -- | Access provided metadata from anywhere
 instance MonadMetadata Compiler where
     getMetadata = compilerGetMetadata
-    getMatches  = compilerGetMatches
+    getMatches  = compilerGetMatches KindContent
+    getAllMetadata pattern = do
```

Comment:
~Name suggestion: `getMatchingMetadata`?~ I thought this PR introduced this function but that's not the case., apologies